### PR TITLE
Add GET /preparations/payments

### PR DESCRIPTION
**Pull request by intelliot**
## High Level Overview of Change

There are several different ways to perform reliable transaction submission. Each has various trade-offs. The one that I recommend for most use cases can be simplified to a three-step process:

1. Prepare the transaction: This is the most important step. The Fee, Sequence, and LastLedgerSequence must be properly determined. XRP-API should "pre-flight" the transaction, if possible, to determine whether it is likely to succeed. The application must persist the transaction in order to have the ability to recover from a power outage and ensure idempotency in step 2.
2. Sign and submit the transaction
3. Get the transaction's final status

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

Also, this PR changes the path prefix to /v3 in order to distinguish this API from others:
  - ripple-rest: /v1
  - Data API v2: /v2

## Test Plan

Will write a script that creates and submits transactions using the new API endpoints.

### CODE REVIEW

File: src/api-v1/paths/preparations/payments.ts [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+      const response = {
+        message,
+        errors: [error]
+      };
+      finishRes(res, status, response);
+      return;
+    }
+
+    if (options.destination) {
+      const destination = getClassicAccountAndTag(options.destination); // TODO: test/handle invalid destination
```
Comment (austin-king) [inline src/api-v1/paths/preparations/payments.ts:80]: Are you comfortable merging before this validation is built?

File: src/api-v1/paths/preparations/payments.ts [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+      if (destinationHasNoTag || amountLessThanBaseReserve) {
+
+        // The balance in XRP drops, or `undefined` if the account does not exist.
+        const {
+          destinationAccountBalance,
+          destinationRequiresTag
+        }: {
+          destinationAccountBalance?: string,
+          destinationRequiresTag?: boolean
+          } = await api.request('account_info', { // TODO: test
```
Comment (austin-king) [inline src/api-v1/paths/preparations/payments.ts:95]: Same with this TODO, was this intentionally left out of scope of this PR?

File: src/api-v1/paths/preparations/payments.ts [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+ * @returns {ClassicAccountAndTag}
+ *          The classic account and tag.
+ */
+function getClassicAccountAndTag(
+  Account: string,
+  expectedTag?: number
+): ClassicAccountAndTag {
+  if (isValidXAddress(Account)) {
+    const classic = xAddressToClassicAddress(Account)
+    if (expectedTag !== undefined && classic.tag !== expectedTag) {
```
Comment (austin-king) [inline src/api-v1/paths/preparations/payments.ts:255]: && -> || ?

Comment (intelliot) [reply src/api-v1/paths/preparations/payments.ts:255]: && is intentional in this case; if there is no `expectedTag`, then there is no need to check that it matches `classic.tag`

**Pull request merged**